#Analysis of assemblies re-arragements by TRE. 

Promer alignment of Assemblies. Assemblies constructed by Jordan at NIAB.

1. Alignment of our assemblies against the reference A22 C. albicans

```bash
Reference=$(ls assembly/misc_publications/c.albicans/A22_Chromosomes_29.fasta)
for Query in $(ls niab_assemblies/c.albicans/UoK_July2023/assemblies/*/*.fasta); do
Strain=$(echo $Query | rev | cut -f2 -d '/' | rev)
Organism=$(echo $Query | rev | cut -f5 -d '/' | rev)
echo "$Organism - $Strain"
Prefix="$Strain"_vs_A22
OutDir=analysis/genome_alignment/mummer/$Organism/$Strain/$Prefix
ProgDir=git_repos/tools/seq_tools/genome_alignment/MUMmer
sbatch $ProgDir/sub_nucmer_slurm.sh $Reference $Query $Prefix $OutDir
done
```

2. Coverage of Nanopore reads over Assembly. For the alignment of ONT reads versus Nanopore assembly use the program minimap:

```bash
for Assembly in $(ls niab_assemblies/c.albicans/UoK_July2023/assemblies/*/*.fasta); do
Reference=$(ls assembly/misc_publications/c.albicans/A22_Chromosomes_29.fasta)
Strain=$(echo $Assembly | rev | cut -f2 -d '/'| rev)
Organism=$(echo $Reference | rev | cut -f2 -d '/' | rev)
Reads=$(ls niab_assemblies/c.albicans/UoK_July2023/porechop/$Strain/*.fastq.gz)
Prefix="${Organism}_${Strain}"
OutDir=analysis/genome_alignment/minimap/$Organism/vs_${Strain}
mkdir $OutDir
ProgDir=git_repos/tools/seq_tools/genome_alignment
sbatch $ProgDir/minimap/slurm_minimap2_2.sh $Reference $Reads $OutDir
done
```

3. SNP Calling of strains.

3.1 First create  a custom SnpEff genome database as SC5314 has still not been created as a reference. 

```bash
SnpEff=/home/sv264/local/bin/snpEff
nano $SnpEff/snpEff.config
```
```
Add the following lines to the section with databases:
#---
# EMR Databases
#----
# Fus2 genome
Fus2v1.0.genome : Fus2
# Bc16 genome
Bc16v1.0.genome: BC-16
# P414 genome
P414v1.0.genome: 414
# 62471 genome
62471v1.0.genome: 62471
# R36_14 genome
R36_14v1.0.genome: R36_14
# SCRP371 genome
SCRP371v1.0.genome: SCRP371
# P. stipis
Ps589v1.0.genome: 589
PsY-11545v1.0.genome: Y-11545
PsCBS6054v1.0.genome: CBS6054
PsAB108.genome: ab1082
PsAB918.genome: ab918
# C. albicans 
Calbicans.genome: SC5314
```

Collect input files

```bash
Organism="C.albicans"
Strain="SC5314"
DbName="Calbicans"
ProjDir=$PWD
Reference=$(ls $ProjDir/assembly/misc_publications/c.albicans/A22_Chromosomes_29.fasta)
Gff=$(ls $ProjDir/assembly/misc_publications/c.albicans/*.gff)
SnpEff=/home/sv264/local/bin/snpEff
mkdir $SnpEff/data/${DbName}
cp $Reference $SnpEff/data/${DbName}/sequences.fa
cp $Gff $SnpEff/data/${DbName}/genes.gff

#Build database using GFF3 annotation
java -jar $SnpEff/snpEff.jar build -gff3 -v ${DbName}
```

3.2 SNP Calling of TRE strains.

# AB55 vs C3, C5 and F2. 
#
# Alignment of the strains generated by TRE translocations against the reference SC5314 A22.
# 
# 

```bash
for Reference in $(ls assembly/misc_publications/c.albicans/A22_Chromosomes_29.fasta); do
for StrainPath in $(ls -d raw_dna/novogene/*/*/*/01.RawData/F2); do
Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)
Organism=$(echo $Reference | rev | cut -f2 -d '/' | rev)
Ref=$(echo $Reference | rev | cut -f2 -d '/' | rev)
F_Read=$(ls $StrainPath/*.fq.gz)
R_Read=$(ls $StrainPath/*.fq.gz)
echo $F_Read
echo $R_Read
echo $Ref
Prefix="${Organism}_vs_${Strain}"
echo $Prefix
OutDir=analysis/genome_alignment/bwa/$Organism/$Strain/vs_${Ref}
ProgDir=~/git_repos/tools/seq_tools/genome_alignment/bwa
sbatch $ProgDir/sub_bwa_slurm.sh $Prefix $Reference $F_Read $R_Read $OutDir
done
done
```

Prepare genome reference indexes required by GATK if needed:

```bash
for Reference in $(ls assembly/misc_publications/c.albicans/A22_Chromosomes_29.fasta); do
OutName=$(echo $Reference | sed 's/.fasta/.dict/g')
OutDir=$(dirname $Reference)
ProgDir=~/local/bin/picard-2.27.3
java -jar $ProgDir/picard.jar CreateSequenceDictionary R=$Reference O=$OutName
samtools faidx $Reference
done
```

#Run with the script in sbtacth_run 

#After vcf is generated filter SNPs:


#SC5314 vs C5

```bash
Vcf=$(ls analysis/popgen/SNP_calling/c.albicans/SC5314/vs_C5/bwagatk_C5.vcf)
vcftools=/home/sv264/local/bin/vcftools_0.1.13/bin
vcflib=/home/sv264/miniconda3/pkgs/vcflib-1.0.0_rc2-h56106d0_2/bin
mq=40
qual=30
dp=10
gq=30
na=1.00
removeindel=N
echo "count prefilter"
cat ${Vcf} | grep -v '#' | wc -l

export LD_LIBRARY_PATH=/home/sv264/miniconda3/pkgs/bzip2-1.0.8-h7b6447c_0/lib
export LD_LIBRARY_PATH=/home/sv264/miniconda3/lib
export PYTHONPATH=/usr/lib64/python2.7
$vcflib/vcffilter -f "QUAL > $qual & MQ > $mq" $Vcf \
| $vcflib/vcffilter -g "DP > $dp & GQ > $gq" > ${Vcf%.vcf}_qfiltered.vcf

echo "count qfilter"
cat ${Vcf%.vcf}_qfiltered.vcf | grep -v '#' | wc -l
$vcftools/vcftools --vcf ${Vcf%.vcf}_qfiltered.vcf --max-missing $na --remove-indels --recode --out ${Vcf%.vcf}_qfiltered_presence


$vcftools/vcftools --vcf ${Vcf%.vcf}_qfiltered.vcf --max-missing $na --keep-only-indels --recode --out ${Vcf%.vcf}_indels_qfiltered_presence
```
```
count prefilter
82913

count qfilter
81003

After filtering, kept 1 out of 1 Individuals
Outputting VCF file...
After filtering, kept 68344 out of a possible 81003 Sites
Run Time = 2.00 seconds


After filtering, kept 1 out of 1 Individuals
Outputting VCF file...
After filtering, kept 12557 out of a possible 81003 Sites
Run Time = 0.00 seconds

```

#Collect VCF stats
#General VCF stats (remember that vcftools needs to have the PERL library exported)

```bash
  Isolate="C5"
  VcfTools=/home/sv264/local/bin/vcftools_0.1.13/perl
  export PERL5LIB="$VcfTools:$PERL5LIB"
  VcfFiltered=$(ls analysis/popgen/SNP_calling/c.albicans/SC5314/vs_C5/*_qfiltered_presence*.vcf | grep -v 'indels')
  Stats=$(echo $VcfFiltered | sed 's/.vcf/.stat/g')
  perl $VcfTools/vcf-stats $VcfFiltered > $Stats
  VcfFiltered=$(ls analysis/popgen/SNP_calling/c.albicans/SC5314/vs_C5/*_qfiltered_presence*.vcf | grep 'indels')
  Stats=$(echo $VcfFiltered | sed 's/.vcf/_indels.stat/g')
  perl $VcfTools/vcf-stats $VcfFiltered > $Stats
```
Calculate the index for percentage of shared SNP alleles between the individuals.

```bash
Isolate="C5"
  for Vcf in $(ls analysis/popgen/SNP_calling/c.albicans/SC5314/vs_C5/*_qfiltered_presence*.vcf | grep -v 'indels'); do
      ProgDir=~/git_repos/scripts/popgen/snp
      python2.7 $ProgDir/similarity_percentage.py $Vcf
  done
```








